client/src/pages/Background.js

```diff
@@ -0,0 +1,286 @@
+    display: "block",
+    maxWidth: "100%",
+    maxHeight: "100%",
+  },
+  text: {
+    padding: ".5rem 20% .5rem 20%",
+  },
+  account: {
+    paddingTop: "0.5rem",
+  },
+  title: {
+    color: "#0000ff",
+    fontFamily: "Open Sans",
+    marginTop: "3rem",
+    paddingBottom: "1rem",
+  },
+  nextStep: {
+    borderRadius: 35,
+    margin: "3rem 0rem 0rem 0rem",
+    padding: ".5rem 1.5rem .5rem 1.5rem",
+  },
+  question: {
+    margin: "2rem 0 .5rem 0",
+  },
+});
+const labels = {
+  //array for Rating component
+  1: "JUNIOR",
+  2: "ADVANCED JUNIOR",
+  3: "INTERMEDIATE",
+  4: "SENIOR",
+  5: "ADVANCED SENIOR",
+};
+const details = {
+  //array for Rating component
+  1: "Do not have any experience on interviews",
+  2: "Had one or two interviews before",
+  3: "Had a few job interviews, but need more practice",
+  4: "Had seven to twelve interviews before, and want to know more about interview skills",
+  5: "Had more than twelve interviews before, and know much about interview skills",
+};
+const selectLists = [
+  ["React", "JavaScript", "Node.js", "Angular", "Python"],
+  [
+    "Less than 1 year",
+    "1 to 3 years",
+    "4 to 6 years",
+    "7 to 10 years",
+    "More than 10 years",
+  ],
+];
+
+function Background(props) {
+  //the function component for the background page is declared here
+  const { user, setUser } = useContext(UserContext);
+  console.log(user);
+  const [openSnack, setOpenSnack] = useState(false); //openSnack is one of the states. setOpenSnack is its method to change it. False is default value.
+  const [severity, setSeverity] = useState("error");
+  const [language, setLanguage] = useState("React");
+  const [experience, setExperience] = useState("");
+  const [value, setValue] = useState(2);
+  const [hover, setHover] = useState(-1);
+/*
+  if (user === "failed to fetch" || user === null) {
+    return <Redirect to="/login" />;
+  } else if (user.backgroundCompleted === true) {
+    return <Redirect to="/dashboard" />;
+  }*/
+
+  if (user === null) {
+    return <p>Loading profile...</p>;
+  } else if (user.backgroundCompleted) {
+    props.history.push({pathname: "/dashboard"});
+  }
+
+  const handleClick = () => {
+    //the method to handle "next step" button being clicked
+    if (experience === "") {
+      setOpenSnack(true);
+    } else {
+      user.backgroundCompleted = true;
+      user.background.language = language;
+      user.background.experience = experience;
+      user.background.rating = value;
+      setUser(user);
+      console.log(user.background);
+      //fetch here
+      const res = fetch("/" + user.id, {
+        method: "put",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify(user),
+      })
+        .then((response) => response.json())
+        .then((responseJson) => {
+          console.log(responseJson);
+          return responseJson;
+        })
+        .catch((error) => {
+          console.error(error);
```

Comment:
Did you want to show an error message using the snackbar if there is a failure for some reason?

Reply:
Good idea. Let us show the error with the snackbar

Reply:
.catch((error) => {
          console.error(error);
          setMessage("Database connection problem!");
          setOpenSnack(true);
        });

Reply:
Good, remove the commented out code though please

Reply:
removed